### PR TITLE
Fix dataframe arithmetics for columns having several value buffers (column size is more than 2 Gb)

### PR DESCRIPTION
**Pull request by asmirnov82**
Arithmetics for columns containg multiple buffers (actualy with size more than 2 Gb) was incorrect. Buffer number was not taken into account

### CODE REVIEW

**Review by JakeRadMSFT — APPROVED** [review]

**Comment by JakeRadMSFT** [comment]
Thanks @asmirnov82! Can you merge this into https://github.com/JakeRadMSFT/machinelearning/tree/u/jakerad/generic-math as well?

File: src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnArithmetic.cs [hunk 1]
```diff
@@ -307,47 +307,51 @@ public void RightShift(PrimitiveColumnContainer<bool> column, int value)
...
         {
+            long index = 0;
             for (int b = 0; b < left.Buffers.Count; b++)
             {
                 var span = left.Buffers[b].ReadOnlySpan;
                 var otherSpan = right.Buffers[b].ReadOnlySpan;
                 for (int i = 0; i < span.Length; i++)
                 {
-                    ret[i] = (span[i] == otherSpan[i]);
+                    ret[index++] = (span[i] == otherSpan[i]);
```
Comment (JakeRadMSFT) [inline src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnArithmetic.cs:317]: Wow - this was really broken.

Comment (JakeRadMSFT) [reply src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnArithmetic.cs:317]: Thanks for catching this.

**Comment by asmirnov82** [comment]
> Thanks @asmirnov82! Can you merge this into https://github.com/JakeRadMSFT/machinelearning/tree/u/jakerad/generic-math as well?
> 
> 
> It's possible this was fixed by other changes in the generic math branch. Can you make sure it meets your needs/add/update tests?

Hello Jake, currently I am on vacation, will do this, when I return

**Comment by JakeRadMSFT** [comment]
> > Thanks @asmirnov82! Can you merge this into https://github.com/JakeRadMSFT/machinelearning/tree/u/jakerad/generic-math as well?
> > It's possible this was fixed by other changes in the generic math branch. Can you make sure it meets your needs/add/update tests?
> 
> Hello Jake, currently I am on vacation, will do this, when I return

Awesome! I merged in all your other changes into the branch. 

We're working on creating better branching and process for handling changes between current train (ML.NET 3.0) and next release train (ML.NET 4.0). Hopefully we'll have it figured out by the time you return!

Thanks again for all these contributions!!!

**Comment by JakeRadMSFT** [comment]
Merged into Generic Math branch!

**Pull request merged**